### DISCUSSION TITLE
[SOLVED] [newbie] Host shows twice in Windows' NetHood; wsdd resolves names through broadcast?

### DISCUSSION THREAD

**Discussion by Shohreh** [christgau/wsdd#164] [Q&A] [answered]
Hello,

I just installed wsdd on Debian for the first time, and have a couple of newbie questions about using it with Samba:

1. When working from a Windows7 client, the host shows twice in Windows' Network Neighborhood (/etc/hosts + smb.conf) while it only shows once when running Windows' "net view" CLI command. How can I prevent it showing twice in the GUI?

2. With WINS, hosts could register to avoid broadcasting, while it looks like wsdd only works through broadcast. Is there a way for a wsdd on the LAN to act as name server?

Thank you.

**Comment by christgau** [comment, marked as answer]
>1. When working from a Windows7 client, the host shows twice in Windows' Network Neighborhood (/etc/hosts + smb.conf) while it only shows once when running Windows' "net view" CLI command. How can I prevent it showing twice in the GUI?

It's very likely that the NETBIOS client of Windows 7 is active and thus the OS discovers your Samba host both with the legacy method and the WSD protocol. You can check that by adding the "Discovery Method" column in the detailed view of the Explorer when you are on the "Network" view. It should show different methods for the two hosts with the same name. 

In that case you can disable one of the two methods. So, either disable wsdd 🙁😉 or disable Netbios in Samba (think it is the nmb [?] daemon). 

>2. With WINS, hosts could register to avoid broadcasting, while it looks like wsdd only works through broadcast. 

Well, multicast but in principle yes, there is no "register" function in the WSD protocol and thus wsdd. Regardless of the protocol, where should the service/device register itself?

> Is there a way for a wsdd on the LAN to act as name server?

That appears to be unrelated to the above broadcast question. Nevertheless, the WSD protocol does not provide a name resolution service, and so wsdd does not have it as well. It is something that the LLMNR protocol provides. wsdd2 over here at Github implements this, AFAIK. And IIRC, systemd has it as well, but it must be enabled explicitly.

Does this help?

**Reply by Shohreh** [reply]
Thanks for the help.

Indeed, logically enough, it shows twice, once through NetBIOS (nmbd) and a second time through WSD (wsdd):

![image](https://user-images.githubusercontent.com/1541861/211768063-9402349d-f631-4d60-be43-7b20320fed5b.png)

So keeping "net view" (through Samba's nmbd) means getting duplicates in the GUI; Turning it off means losing "net view" :-/ Bah, I can live with that.

So without WINS, also provided by nmbd, hosts advertise themselves through multicast/broadcast. Since it's only used on a LAN anyway, I guess it's a non-issue for all practical purposes, although a dynamic name server instead of multicasting/broadcasting looks good on paper.

On neither Win7 nor Win10 do I see a "Web Services Dynamic Discovery" service: What service provides WSD?

![image](https://user-images.githubusercontent.com/1541861/211768416-92eac62b-5452-4021-b25a-52ca8b45b01d.png)

--
Edit: Apparently, it's not a service. It can be turned on/off in Advanced sharing settings.

**Reply by christgau** [reply]
> Turning it [netbios] off means losing "net view" :-/ Bah, I can live with that.

IMO, you'll lose net view anyways. I don't have it anymore on Windows 11 and Windows tells me that it requires the old SMB1 protocol, which is dangerous, insecure, and the like (as stated by the error message). That's why wsdd is here. 😉If you now intent to ask for a command utility that would list all discovered devices/services, I have no idea about such a tool. Maybe there are scriptlets for powershell that do so...

> On neither Win7 nor Win10 do I see a "Web Services Dynamic Discovery" service: What service provides WSD?

AFAIK, it's called "Function Discovery Provider Host"

HTH

PS: Thanks marking the discussion as "solved". If possible please mark the answer as such so that we have this nice green tick mark on this thread 😉

**Reply by Shohreh** [reply]
I don't know where to click to get that green tick. Clicking on the up arrow didn't do it.

**Reply by christgau** [reply]
> I don't know where to click to get that green tick. Clicking on the up arrow didn't do it.

Reload and try again, I think it only works in Q&A discussions.

**Comment by Shohreh** [comment]
I did, It does look like it's not available here.

**Reply by christgau** [reply]
huh, strange: 
![grafik](https://user-images.githubusercontent.com/39031063/211927044-03758d87-ebf2-4ee6-91c7-4194a68ab016.png)

**Reply by Shohreh** [reply]
The page didn't have those buttons when I replied. Weird.
